src/python/pants/pantsd/pants_daemon.py

```diff
@@ -493,6 +494,18 @@ def terminate(self, include_watchman=True):
     if include_watchman:
       self.watchman_launcher.terminate()
 
+  def _has_exceeded_memory_usage(self):
+    used_memory = self.current_memory_usage()
+    humanfriendly_max_memory = self._bootstrap_options.for_global_scope().daemon_max_memory_usage
+    parsed_max_memory = humanfriendly.parse_size(humanfriendly_max_memory)
+    exceeded = used_memory > parsed_max_memory
+    self._logger.debug(f"pantsd is using {used_memory}b of memory, max is {humanfriendly_max_memory} ({parsed_max_memory} bytes)")
```

Comment:
This will be logged in the client's log. Would it be useful to add something similar to the server's log?